fbcnms-packages/fbcnms-ui/components/FormField.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright 2004-present Facebook. All Rights Reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ */
+
+import Box from '@material-ui/core/Box';
+import Divider from '@material-ui/core/Divider';
+import Grid from '@material-ui/core/Grid';
+import HelpIcon from '@material-ui/icons/Help';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import LinearProgress from '@material-ui/core/LinearProgress';
+import ListItem from '@material-ui/core/ListItem';
+import OutlinedInput from '@material-ui/core/OutlinedInput';
+import React from 'react';
+import Tooltip from '@material-ui/core/Tooltip';
+import Typography from '@material-ui/core/Typography';
+import Visibility from '@material-ui/icons/Visibility';
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
+import grey from '@material-ui/core/colors/grey';
+
+import {makeStyles} from '@material-ui/styles';
+import {useState} from 'react';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    display: 'flex',
+    marginBottom: '5px',
+    alignItems: 'center',
+  },
+  heading: {
+    flexBasis: '33.33%',
+    marginRight: '15px',
+    textAlign: 'right',
+  },
+  subheading: {
+    fontWeight: '400',
+  },
+  optionalLabel: {
+    color: grey.A700,
+    fontStyle: 'italic',
+    fontWeight: '400',
+    marginLeft: '8px',
+  },
+  secondaryHeading: {
+    flexBasis: '66.66%',
+  },
+  icon: {
+    marginLeft: '5px',
+    paddingTop: '4px',
+    verticalAlign: 'bottom',
+    width: '15px',
```

Comment:
Pixel sizes should be specified as even numbers, by convention a multiple of 4px is helpful. Otherwise, we might end up finding that layouts don't fit properly into different screens, which invariably have widths and heights with pixel counts that are multiples of 4.
```suggestion
    width: '16px',
```